src/WarcraftLegacies.Source/Quests/Scourge/QuestPlague.cs

```diff
@@ -20,59 +24,50 @@ namespace WarcraftLegacies.Source.Quests.Scourge
       "Plague of Undeath",
       "The Cult of the Damned is prepared to unleash a devastating zombifying plague across the lands of Lordaeron.",
       @"ReplaceableTextures\CommandButtons\BTNPlagueBarrel.blp")
     {
-      _gilneasDoor = preplacedUnitSystem.GetUnit(Constants.UNIT_H02K_GREYMANE_S_GATE_CLOSED).SetInvulnerable(true);
       _plagueVictim = plagueVictim;
       _plagueParameters = plagueParameters;
       _secondaryPlagueFaction = secondaryPlagueFaction;
       AddObjective(new ObjectiveEitherOf(
         new ObjectiveResearch(Constants.UPGRADE_R06I_PLAGUE_OF_UNDEATH_SCOURGE, FourCC("u000")),
         new ObjectiveTime(660)));
-      AddObjective(new ObjectiveTime(540));
+      AddObjective(new ObjectiveTime(420));
+      _deathknellUnits = deathknell.PrepareUnitsForRescue(RescuePreparationMode.HideAll);
+      _scholomanceUnits = scholomance.PrepareUnitsForRescue(RescuePreparationMode.HideAll);
+      _coastUnits = coast.PrepareUnitsForRescue(RescuePreparationMode.HideAll);
       Global = true;
       Required = true;
       ResearchId = Constants.UPGRADE_R009_QUEST_COMPLETED_PLAGUE_OF_UNDEATH;
 
-      _portalController1 = preplacedUnitSystem.GetUnit(Constants.UNIT_N03J_BLACK_PORTAL_AURA_CONTROL_NEXUS, new Point(-4137, 16957)).SetInvulnerable(true);
-      _portalController2 = preplacedUnitSystem.GetUnit(Constants.UNIT_N03J_BLACK_PORTAL_AURA_CONTROL_NEXUS, new Point(14198, 6530)).SetInvulnerable(true);
-      _innerWaygate1 = preplacedUnitSystem.GetUnit(Constants.UNIT_N03H_DEATH_GATE_WAYGATE, Regions.Scholomance_Exterior_1.Center).Show(false);
-      _innerWaygate2 = preplacedUnitSystem.GetUnit(Constants.UNIT_N03H_DEATH_GATE_WAYGATE, Regions.Scholomance_Exterior_2.Center).Show(false);
-      _outerWaygate1 = preplacedUnitSystem.GetUnit(Constants.UNIT_N03H_DEATH_GATE_WAYGATE, Regions.Wrathgate_Portal_1.Center).Show(false);
-      _outerWaygate2 = preplacedUnitSystem.GetUnit(Constants.UNIT_N03H_DEATH_GATE_WAYGATE, Regions.Wrathgate_Portal_2.Center).Show(false);
     }
 
     /// <inheritdoc />
     protected override string RewardFlavour =>
-      "The plague has been unleashed! The citizens of Lordaeron are quickly transforming into mindless zombies, and the Black Gate has been opened in Dragonblight.";
+      "The plague has been unleashed! The citizens of Lordaeron are quickly transforming into mindless zombies";
 
     /// <inheritdoc />
     protected override string RewardDescription =>
-      "All villagers in Lordaeron are transformed into Zombies, several Zombie-spawning Plague Cauldrons spawn throughout Lordaeron, a portal opens between Dragonblight and Scholomance, you learn to build Necropoli, and Lordaeron's Control Points reset to level 0";
+      "All villagers in Lordaeron are transformed into Zombies, several Zombie-spawning Plague Cauldrons spawn throughout Lordaeron, three bases around Lordaeron and Lordaeron's Control Points reset to level 0";
```

Comment:
Doesn't seem to indicate that you get a choice about where to invade, along with a free teleport. Is that shown somewhere else?

Reply:
Its shown on the research that enables the dialogue!

Reply:
Cool. But the way it's actually set up, the quest gives you the Invasion, not the research. You can even complete the quest without doing the research at all by just waiting until turn 11. Even if that wasn't the case, everything _else_ the quest gives you is written on the quest, so this should be on the quest too.

Reply:
allright!